### PR TITLE
Polys hash set

### PR DESCRIPTION
**Pull request by certik**
@rwst merged the work in #597 and #578.

### CODE REVIEW

File: symengine/eval_arb.cpp [hunk 1]
```diff
@@ -130,6 +131,12 @@ class EvalArbVisitor : public BaseVisitor<EvalArbVisitor> {
         throw std::runtime_error("Not implemented.");
     }
 
+    virtual void visit(const Polynomial &x) {
```
Comment (isuruf) [inline symengine/eval_arb.cpp]: You can now change this to `void bvisit(const Polynomial &x) {`

File: symengine/eval_arb.cpp [hunk 2]
```diff
@@ -130,6 +131,12 @@ class EvalArbVisitor : public BaseVisitor<EvalArbVisitor> {
...
+        throw std::runtime_error("Not implemented.");
+    }
+
+    virtual void visit(const Complex &) {
```
Comment (isuruf) [inline symengine/eval_arb.cpp]: This is something left over from merge

File: symengine/polys.cpp [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+vec_basic Polynomial::get_args() const
+{
+    vec_basic args;
+    return args;
+}
+
+//TODO
+RCP<const Basic> Polynomial::diff(const RCP<const Symbol> &x) const
+{
+    return x;
```
Comment (certik) [inline symengine/polys.cpp]: This should raise an exception before it gets implemented.

File: symengine/polys.cpp [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+    #else
+    throw::std::runtime_error("Not Implemented");
+    #endif
+}
+
+//TODO
+vec_basic Polynomial::get_args() const
+{
+    vec_basic args;
+    return args;
```
Comment (certik) [inline symengine/polys.cpp]: This should raise an exception before it gets implemented.

**Comment by rwst** [comment]
gdb backtrace:
```
#0  piranha::kronecker_array<long long>::decode<std::vector<long long, std::allocator<long long> > > (retval=std::vector of length 0, capacity 0, 
    n=@0x7ffff2f420b0: 4) at /usr/local/include/piranha/kronecker_array.hpp:326
#1  0x00000000006f3a6f in SymEngine::StrPrinter::bvisit (this=0x7fffffffc450, 
    x=...) at /home/ralf/symengine/symengine/printer.cpp:330
#2  0x0000000000634445 in SymEngine::BaseVisitor<SymEngine::StrPrinter>::visit (
    this=0x7fffffffc450, x=...)
    at /home/ralf/symengine/symengine/type_codes.inc:21
#3  0x00000000006ebb86 in SymEngine::Polynomial::accept (this=0xa416e0, v=...)
    at /home/ralf/symengine/symengine/type_codes.inc:21
#4  0x00000000006f51d8 in SymEngine::StrPrinter::apply (this=0x7fffffffc450, 
    b=...) at /home/ralf/symengine/symengine/printer.cpp:509
#5  0x0000000000631966 in SymEngine::Basic::__str__ (this=0xa416e0)
    at /home/ralf/symengine/symengine/basic.cpp:33
#6  0x0000000000600863 in ____C_A_T_C_H____T_E_S_T____238 ()
    at /home/ralf/symengine/symengine/tests/basic/test_polynomial.cpp:272
```

**Comment by Sumith1896** [comment]
With Piranha poly now available, this PR doesn't make much sense. Closing this.

**Pull request closed without merging**